### PR TITLE
docs(python): add note to connectorx in user-guide advising arm can't pip install

### PR DESCRIPTION
**Pull request by deanm0000**
https://github.com/pola-rs/polars/issues/14138

### CODE REVIEW

File: docs/user-guide/io/database.md [hunk 1]
```diff
@@ -37,6 +37,8 @@ To read from one of the supported databases with `ConnectorX` you need to activa
 $ pip install connectorx
 ```
 
+Note: connectorx does not have wheels for ARM architecture and won't work as a pip install. In that scenario, you must build the package yourself. See [here](https://github.com/sfu-db/connector-x/issues/186) for more details.
```
Comment (mcrumiller) [inline docs/user-guide/io/database.md:40]: Suggested wording ("in that scenario" flows a bit oddly):

> Note: `connectorx` cannot be installed on ARM architectures from pip. See [this thread](https://github.com/sfu-db/connector-x/issues/186) for manual build instructions.

Comment (deanm0000) [reply docs/user-guide/io/database.md]: I agree, that's better.

**Comment by alexander-beedie** [comment]
Actually it looks like connectorx _can_ be installed on ARM64 - their build targets include `aarch64-unknown-linux-gnu` and `aarch64-apple-darwin` (aka: Apple Silicon, which is itself an ARM64 derivative), so let's not merge this until the exact issue is better understood.

**Comment by stinodego** [comment]
> I'm also not even that sure we should be providing custom build instructions for (potentially) every optional package we reference, as this is a generic issue for less-common architectures, rather than being related to Polars itself. Presumably there are more packages that won't compile, and I can't see that it's on us to provide build instructions for all of them... (@stinodego, want to weigh-in on this one?)

This also feels out-of-scope for me.

**Comment by deanm0000** [comment]
I just learned this word https://thedecisionlab.com/biases/bikeshedding

**Comment by stinodego** [comment]
Thanks @deanm0000 for the initiative here, but this information feels a bit too specific to be included in the user guide. It's also subject to becoming outdated whenever connectorx would decide to add linux aarch wheels. So I'm closing this one.

**Pull request closed without merging**